infrastructure/src/main/java/org/corfudb/infrastructure/log/SlidingWindow.java

```diff
@@ -0,0 +1,54 @@
+package org.corfudb.infrastructure.log;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.ArrayList;
+import java.util.concurrent.TimeUnit;
+
+@Slf4j
+public class SlidingWindow {
+    private long cntWindow;
+    private long timeWindow;
+    private long threshVal; //this is in seconds
+    private long startTime = 0;
+
+    @Getter
+    ArrayList<Long> intervals;
+
+    public SlidingWindow(long cntWindow, long timeWindow, long threshval) {
+        this.cntWindow = cntWindow;
+        this.timeWindow = timeWindow;
+        this.threshVal = threshval;
+        startTime = 0;
+        intervals = new ArrayList<>();
```

Comment:
Why is this array needed?

You just need two variables, startOfWindow and spikeCounter. Also, a circular buffer is probably more efficient than this, since a remove needs to do a shift by left op.